src/document_parser_callbacks.h

```diff
@@ -27,37 +27,40 @@ really_inline error_code parser::on_success(error_code success_code) noexcept {
 }
 really_inline bool parser::on_start_document(uint32_t depth) noexcept {
   containing_scope_offset[depth] = current_loc;
+  containing_scope_count[depth] = 0;
   write_tape(0, internal::tape_type::ROOT);
   return true;
 }
 really_inline bool parser::on_start_object(uint32_t depth) noexcept {
   containing_scope_offset[depth] = current_loc;
+  containing_scope_count[depth] = 0;
   write_tape(0, internal::tape_type::START_OBJECT);
   return true;
 }
 really_inline bool parser::on_start_array(uint32_t depth) noexcept {
   containing_scope_offset[depth] = current_loc;
+  containing_scope_count[depth] = 0;
   write_tape(0, internal::tape_type::START_ARRAY);
   return true;
 }
 // TODO we're not checking this bool
 really_inline bool parser::on_end_document(uint32_t depth) noexcept {
   // write our doc.tape location to the header scope
   // The root scope gets written *at* the previous location.
-  annotate_previous_loc(containing_scope_offset[depth], current_loc);
+  annotate_previous_loc(containing_scope_offset[depth], current_loc, 0);
```

Comment:
Since it's clear that the only places that use annotate_previous_loc() are containers, maybe annotate_previous_loc() could just be something like finalize_container(uint32_t depth), and automatically take these arguments? Not necessary at all, it's legacy--and it's very subject to taste.

Reply:
I agree.

Reply:
Addressed in a subsequent commit.